clang/test/SemaSYCL/named_sub_group_size.cpp

```diff
@@ -0,0 +1,48 @@
+// RUN: %clang_cc1 -internal-isystem %S/Inputs -fsycl-is-device -sycl-std=2020 -fsyntax-only -ast-dump -verify=expected,primary %s | FileCheck %s
```

Comment:
Why is `-verify=expected,primary` needed here? Isn't `-verify `enough?

Reply:
>Isn't `-verify `enough?

No, it is not enough.

> Why is `-verify=expected,primary` needed here? 
We need `-verify=expected,primary`  because of the diagnostic check here. 
```
[[intel::named_sub_group_size(primary)]] void AttrFunc() {} // #AttrFunc

// Test attribute does not get propagated to the kernel.
void calls_kernel_3() {
  // CHECK:     FunctionDecl {{.*}}Kernel4
  // CHECK-NOT: IntelNamedSubGroupSizeAttr {{.*}}
  sycl::kernel_single_task<class Kernel4>([]() { // #Kernel4
    // primary-error@#AttrFunc{{kernel-called function must have a sub group size that matches the size specified for the kernel}}
    // primary-note@#Kernel4{{kernel declared here}}
    AttrFunc();
  });
}
```

Reply:
Sorry if I am being dense here, but why can't this just be the 'normal' `expected-error` which would just work with `-verify`

Reply:
Done. Updated test.